### PR TITLE
add integration tests

### PR DESCRIPTION
**Pull request by JAndrewCH**

### CODE REVIEW

File: src/ITest/java/uk/gov/companieshouse/pscstatementdataapi/config/AbstractMongoConfig.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import org.testcontainers.containers.MongoDBContainer;
+import org.testcontainers.utility.DockerImageName;
+
+/**
+ * Mongodb configuration runs on test container.
+ */
+public class AbstractMongoConfig {
+
+    public static final MongoDBContainer mongoDBContainer = new MongoDBContainer(
+            DockerImageName.parse("mongo:4.0.10"));
```
Comment (bcullerton) [inline src/ITest/java/uk/gov/companieshouse/pscstatementdataapi/config/AbstractMongoConfig.java:14]: Should this be hard set in the code? Could we pass in as a system var maybe for when CH upgrades mongo

File: src/ITest/resources/features/psc_statements.feature [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      | companyNumber | statementId                 | result        |
+      | OC421554      | DHTUrJoAuKdXw7zvkreyAm_SoH0 | psc_statement |
+
+
+  Scenario Outline: Processing psc statement information successfully
+
+    Given Psc statements data api service is running
+    When I send a PUT request with payload "<data>" file for company number "<companyNumber>" with statement id "<statementId>"
+    Then I should receive 200 status code
+    And the CHS Kafka API is invoked for company number "<companyNumber>" with statement id "<statementId>"
```
[the remark this answers is not in the document]
Comment (bcullerton) [reply src/ITest/resources/features/psc_statements.feature:21]: Same thought for other 200 tests/delete tests

File: src/ITest/java/uk/gov/companieshouse/pscstatementdataapi/steps/PscStatementsSteps.java [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+    public void nothing_persisted_to_database() {
+        List<PscStatementDocument> pscDocs = pscStatementRepository.findAll();
+        Assertions.assertThat(pscDocs).hasSize(0);
+    }
+
+    @After
+    public void dbStop(){
+        mongoDBContainer.stop();
+    }
+}
```
Comment (bcullerton) [inline src/ITest/java/uk/gov/companieshouse/pscstatementdataapi/steps/PscStatementsSteps.java:274]: Like this as a whole although it is a bit of a monster file. Totally optional but we could split off situation, action and confirmation steps and have a given, when and then file or something. Maybe one to look at again when more tests are added

Comment (JAndrewCH) [reply src/ITest/java/uk/gov/companieshouse/pscstatementdataapi/steps/PscStatementsSteps.java:274]: I did think about it but the majority of the steps are used in multiple files so I think we end up with a common steps that is almost as big

File: src/ITest/java/uk/gov/companieshouse/pscstatementdataapi/config/AbstractMongoConfig.java (hunk 1 above)
Comment (JAndrewCH) [reply src/ITest/java/uk/gov/companieshouse/pscstatementdataapi/config/AbstractMongoConfig.java:14]: Yeah I wasn't totally sure but it seems to be in the ITests for basically every project in CH so I figured it was probs easier if we set it the same way everywhere

File: src/ITest/java/uk/gov/companieshouse/pscstatementdataapi/steps/PscStatementsSteps.java (hunk 3 above)
Comment (JAndrewCH) [reply src/ITest/java/uk/gov/companieshouse/pscstatementdataapi/steps/PscStatementsSteps.java:274]: And then have to move steps between files if we add more tests

File: src/ITest/java/uk/gov/companieshouse/pscstatementdataapi/steps/PscStatementsSteps.java [hunk 4]
```diff
@@ -256,6 +258,11 @@ public void statement_exists(String statementId) {
         Assertions.assertThat(pscStatementRepository.existsById(statementId)).isTrue();
     }
 
+    @When("no statement exists with id {string}")
+    public void no_statement_exists(String statementId) {
+        Assertions.assertThat(pscStatementRepository.existsById(statementId)).isTrue();
+    }
+
```
Comment (bcullerton) [inline src/ITest/java/uk/gov/companieshouse/pscstatementdataapi/steps/PscStatementsSteps.java:265]: Should this not be isFalse? Also it's more of an assertion, should it be a Then step??

**Review by bcullerton — APPROVED** [review]

**Pull request merged**